server/controllers/users/getUserChannels.js

```diff
@@ -9,7 +16,19 @@ exports.getUserChannels = (req, res, next) => {
       channelIds = rows.map(({ channel_id: channelId }) => channelId);
       return getChaMessagesByChaId(channelIds);
     })
-    .then(({ rows: messages }) => formatMessages(messages, channelIds))
-    .then((result) => res.json(result))
+    .then(({ rows: messages }) => {
+      const userIds = new Set();
+      messages.forEach((msg) => {
+        userIds.add(msg.user_id);
+      });
+      return Promise.all([
+        formatMessages(messages, channelIds),
+        getUsersByIds(Array.from(userIds)),
+        getLanguages(),
```

Comment:
both the languages and interests are very good examples that database normalization is not always worth it :) They complicate the queries tremendously - 4 additional joins to fully select a user (as there is a user-language table and user-interest table). Instead, probably would be better to store them in a text column in the user table with a full text search index and use the full text search for matching users with same languages / interests.. well, too late now :)